optuna/integration/allennlp.py

```diff
@@ -157,22 +311,47 @@ class AllenNLPPruningCallback(EpochCallback):
     if you want to add a proning callback which observes a metric.
 
     Args:
-        trial:
-            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
-            objective function.
         monitor:
             An evaluation metric for pruning, e.g. ``validation_loss`` or
             ``validation_accuracy``.
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
     """
 
-    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+    def __init__(
+        self,
+        trial: Optional[optuna.trial.Trial] = None,
```

Comment:
Making `trial` and `monitor` keyword argument so that `AllenNLPPruningCallback` can be instantiated without them. This change is introduced for making it possible to create a callback with environment variables. (https://github.com/optuna/optuna/pull/1772/files#diff-194650369fb014d2ea8f428774c04276R336-R350)

Reply:
We should probably explain in the documentation that these arguments are types optional but are mandatory.

Reply:
I added some explanations in 603308d. I'd like your feedback. :bow:

Reply:
Sorry if I wasn't clear but I meant in the user facing documentation.

Reply:
Thank you, it makes sense. :pray:
78822da I added a note in the docstring based on comments in the codebase.